src/Microsoft.AspNetCore.Blazor/Components/EventHandlerInvoker.cs

```diff
@@ -7,49 +7,53 @@
 namespace Microsoft.AspNetCore.Blazor.Components
 {
     /// <summary>
-    /// A bound event handler delegate.
+    /// Provides the ability to invoke an event handler.
     /// </summary>
-    public readonly struct EventHandlerInvoker
+    public readonly struct EventHandlerInvoker<TArg>
     {
         private readonly MulticastDelegate _delegate;
 
-        /// <summary>
-        /// Creates the new <see cref="EventHandlerInvoker"/>.
-        /// </summary>
-        /// <param name="delegate">The delegate to bind.</param>
-        public EventHandlerInvoker(MulticastDelegate @delegate)
+        // Internal because only the framework needs to create these. External code should
+        // use ComponentEvents.InvokeEventHandler, supplying the delegate.
+        internal EventHandlerInvoker(MulticastDelegate @delegate)
         {
             _delegate = @delegate;
         }
+
         /// <summary>
-        /// Invokes the delegate associated with this binding.
+        /// Invokes the associated event handler delegate.
         /// </summary>
-        /// <param name="e">The <see cref="UIEventArgs"/>.</param>
+        /// <param name="eventArg">The argument for the event handler.</param>
         /// <returns></returns>
-        public Task Invoke(UIEventArgs e)
+        public Task Invoke(TArg eventArg)
         {
             switch (_delegate)
             {
                 case Action action:
                     action.Invoke();
                     return Task.CompletedTask;
 
-                case Action<UIEventArgs> actionEventArgs:
-                    actionEventArgs.Invoke(e);
+                case Action<TArg> actionEventArgs:
+                    actionEventArgs.Invoke(eventArg);
                     return Task.CompletedTask;
 
                 case Func<Task> func:
                     return func.Invoke();
 
-                case Func<UIEventArgs, Task> funcEventArgs:
-                    return funcEventArgs.Invoke(e);
+                case Func<TArg, Task> funcEventArgs:
+                    return funcEventArgs.Invoke(eventArg);
 
                 case MulticastDelegate @delegate:
-                    return @delegate.DynamicInvoke(e) as Task ?? Task.CompletedTask;
+                    return @delegate.DynamicInvoke(eventArg) as Task ?? Task.CompletedTask;
 
                 case null:
                     return Task.CompletedTask;
             }
         }
+
+        internal bool CheckDelegateTarget(object target)
```

Comment:
IsSameTarget? I would expect a method named like this to throw

Reply:
That's reasonable. I'll change it.